arelle/plugin/validate/ESEF_2022/Util.py

```diff
@@ -28,6 +32,7 @@
 
 
 _: TypeGetText  # Handle gettext
+imgDataMediaBase64Pattern = re.compile(r"data:image([^,;]*)(;base64)?,(.*)$", re.S)
```

Comment:
For href, it would be good allow [SVG fragments](https://www.w3.org/TR/SVG/linking.html#SVGFragmentIdentifiers) (i.e., trim rather than try/fail to base64 decode).

Reply:
Thank you for the suggestion.

I think that ESEF is more restrictive than bare SVG.

I'll check if SVG fragments are compatible with the ESEF manual.

Reply:
Fyi, noticed a bug around this at any rate. As Brett alluded to this regex captures any SVG fragments at the end of an svg image. For this image:

![arelle](https://user-images.githubusercontent.com/17255946/231516847-ef0b1e78-d244-4f79-b19b-31c4ea9b2b7a.svg)

if I include a fragment to preserve aspect ratio after first base64 encoding

```python
image = 'PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjwhLS0gQ3JlYXRlZCB3aXRoIElua3NjYXBlIChodHRwOi8vd3d3Lmlua3NjYXBlLm9yZy8pIC0tPgoKPHN2ZwogICB4bWxuczpkYz0iaHR0cDovL3B1cmwub3JnL2RjL2VsZW1lbnRzLzEuMS8iCiAgIHhtbG5zOmNjPSJodHRwOi8vY3JlYXRpdmVjb21tb25zLm9yZy9ucyMiCiAgIHhtbG5zOnJkZj0iaHR0cDovL3d3dy53My5vcmcvMTk5OS8wMi8yMi1yZGYtc3ludGF4LW5zIyIKICAgeG1sbnM6c3ZnPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIgogICB4bWxuczpzb2RpcG9kaT0iaHR0cDovL3NvZGlwb2RpLnNvdXJjZWZvcmdlLm5ldC9EVEQvc29kaXBvZGktMC5kdGQiCiAgIHhtbG5zOmlua3NjYXBlPSJodHRwOi8vd3d3Lmlua3NjYXBlLm9yZy9uYW1lc3BhY2VzL2lua3NjYXBlIgogICB2ZXJzaW9uPSIxLjEiCiAgIGlkPSJzdmcyIgogICB4bWw6c3BhY2U9InByZXNlcnZlIgogICB3aWR0aD0iNTM2LjM2NDc1IgogICBoZWlnaHQ9IjE1MC4zMTAwMyIKICAgdmlld0JveD0iMCAwIDUzNi4zNjQ3MyAxNTAuMzEwMDMiCiAgIHNvZGlwb2RpOmRvY25hbWU9ImFyZWxsZS1sb2dvLnN2ZyIKICAgaW5rc2NhcGU6dmVyc2lvbj0iMC45Mi41ICgyMDYwZWMxZjlmLCAyMDIwLTA0LTA4KSI+PG1ldGFkYXRhCiAgICAgaWQ9Im1ldGFkYXRhOCI+PHJkZjpSREY+PGNjOldvcmsKICAgICAgICAgcmRmOmFib3V0PSIiPjxkYzpmb3JtYXQ+aW1hZ2Uvc3ZnK3htbDwvZGM6Zm9ybWF0PjxkYzp0eXBlCiAgICAgICAgICAgcmRmOnJlc291cmNlPSJodHRwOi8vcHVybC5vcmcvZGMvZGNtaXR5cGUvU3RpbGxJbWFnZSIgLz48ZGM6dGl0bGU+PC9kYzp0aXRsZT48L2NjOldvcms+PC9yZGY6UkRGPjwvbWV0YWRhdGE+PGRlZnMKICAgICBpZD0iZGVmczYiPjxjbGlwUGF0aAogICAgICAgY2xpcFBhdGhVbml0cz0idXNlclNwYWNlT25Vc2UiCiAgICAgICBpZD0iY2xpcFBhdGgxOCI+PHBhdGgKICAgICAgICAgZD0iTSAwLDEyMiBIIDc2OC41MDUgViAwIEggMCBaIgogICAgICAgICBpZD0icGF0aDE2IgogICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvY2xpcFBhdGg+PC9kZWZzPjxzb2RpcG9kaTpuYW1lZHZpZXcKICAgICBwYWdlY29sb3I9IiNmZmZmZmYiCiAgICAgYm9yZGVyY29sb3I9IiM2NjY2NjYiCiAgICAgYm9yZGVyb3BhY2l0eT0iMSIKICAgICBvYmplY3R0b2xlcmFuY2U9IjEwIgogICAgIGdyaWR0b2xlcmFuY2U9IjEwIgogICAgIGd1aWRldG9sZXJhbmNlPSIxMCIKICAgICBpbmtzY2FwZTpwYWdlb3BhY2l0eT0iMCIKICAgICBpbmtzY2FwZTpwYWdlc2hhZG93PSIyIgogICAgIGlua3NjYXBlOndpbmRvdy13aWR0aD0iMTg2NCIKICAgICBpbmtzY2FwZTp3aW5kb3ctaGVpZ2h0PSIxMDI0IgogICAgIGlkPSJuYW1lZHZpZXc0IgogICAgIHNob3dncmlkPSJmYWxzZSIKICAgICBpbmtzY2FwZTp6b29tPSIwLjcyMDIyOTUzIgogICAgIGlua3NjYXBlOmN4PSI3ODguNzIwNTgiCiAgICAgaW5rc2NhcGU6Y3k9Ii04NC40NDIyMDQiCiAgICAgaW5rc2NhcGU6d2luZG93LXg9IjU2IgogICAgIGlua3NjYXBlOndpbmRvdy15PSIyNyIKICAgICBpbmtzY2FwZTp3aW5kb3ctbWF4aW1pemVkPSIxIgogICAgIGlua3NjYXBlOmN1cnJlbnQtbGF5ZXI9ImcxMCIKICAgICBmaXQtbWFyZ2luLXRvcD0iMCIKICAgICBmaXQtbWFyZ2luLWxlZnQ9IjAiCiAgICAgZml0LW1hcmdpbi1yaWdodD0iMCIKICAgICBmaXQtbWFyZ2luLWJvdHRvbT0iMCIgLz48ZwogICAgIGlkPSJnMTAiCiAgICAgaW5rc2NhcGU6Z3JvdXBtb2RlPSJsYXllciIKICAgICBpbmtzY2FwZTpsYWJlbD0iYXJlbGxlLWxvZ28iCiAgICAgdHJhbnNmb3JtPSJtYXRyaXgoMS4zMzMzMzMzLDAsMCwtMS4zMzMzMzMzLC0zLjY1MDgxNDEsMTU5LjMyNzc3KSI+PGcKICAgICAgIGlkPSJnMTIiPjxnCiAgICAgICAgIGlkPSJnMTQiCiAgICAgICAgIGNsaXAtcGF0aD0idXJsKCNjbGlwUGF0aDE4KSI+PGcKICAgICAgICAgICBpZD0iZzIwIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEwOS42Njg5LDY3LjE2MzEpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAtMS41OTgsMi42NTMgLTcuMjkyLDUuOTg3IC04LjgwOCwxNS4zMDcgLTAuODA0LDQuOTQ5IDAuNTY5LDcuOTk0IDEuOTU1LDEyLjE4MSAxLjQ4OCw0LjQ3MSAtMS4wNTksNy4yNDQgLTUuMzgsNy40MDggLTguNjQ0LDAuMzMgLTMxLjkwNywtMi4yMTEgLTU3LjYzOCwxMS4yODggLTEwLjA2Niw1LjI4IC0xMC43NTksMy45MjIgLTcuMDc2LC0yLjU2NiA1LjY0NSwtOS45NTIgMTkuMzc4LC0xNS4zMzQgMzUuNjg3LC0xOC4wMDUgLTQ1LjM5Nyw5LjgyNSAtMTcuMTY2LC0yLjE3NCA1LjA5OCwtNy4xMjcgLTQxLjA1Niw4LjQ4NCAtMTUuNjI1LC0zLjAzNSA0LjQ0OCwtNi43MTEgLTM4LjQ3NSw2LjgyMSAtMTEuMzg3LC0yLjk3OCAxLjQ3NywtNS40MjggMy44NjIsLTAuNzM2IDEyLjM5MSwtMTMuMDI2IDE4LjA1NCwtMTAuMjczIDYuMzE3LDMuMDY5IDQuNDg4LDAuNjg5IDcuNDc5LDIuMDI2IEMgMC41NjYsMC40NTggMCwwIDAsMCIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOiNhYTExOGU7ZmlsbC1vcGFjaXR5OjE7ZmlsbC1ydWxlOm5vbnplcm87c3Ryb2tlOm5vbmUiCiAgICAgICAgICAgICBpZD0icGF0aDIyIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzI0IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEwOS42Njg5LDY3LjE2MzEpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAtMS41OTgsMi42NTMgLTcuMjkyLDUuOTg3IC04LjgwOCwxNS4zMDcgLTAuODA0LDQuOTQ5IDAuNTY5LDcuOTk0IDEuOTU1LDEyLjE4MSAxLjQ4OCw0LjQ3MSAtMS4wNTksNy4yNDQgLTUuMzgsNy40MDggLTguNjQ0LDAuMzMgLTMxLjkwNywtMi4yMTEgLTU3LjYzOCwxMS4yODggLTEwLjA2Niw1LjI4IC0xMC43NTksMy45MjIgLTcuMDc2LC0yLjU2NiA1LjY0NSwtOS45NTIgMTkuMzc4LC0xNS4zMzQgMzUuNjg3LC0xOC4wMDUgLTQ1LjM5Nyw5LjgyNSAtMTcuMTY2LC0yLjE3NCA1LjA5OCwtNy4xMjcgLTQxLjA1Niw4LjQ4NCAtMTUuNjI1LC0zLjAzNSA0LjQ0OCwtNi43MTEgLTM4LjQ3NSw2LjgyMSAtMTEuMzg3LC0yLjk3OCAxLjQ3NywtNS40MjggMy44NjIsLTAuNzM2IDEyLjM5MSwtMTMuMDI2IDE4LjA1NCwtMTAuMjczIDYuMzE3LDMuMDY5IDQuNDg4LDAuNjg5IDcuNDc5LDIuMDI2IEMgMC41NjYsMC40NTggMCwwIDAsMCBaIgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2ZmZmZmZjtzdHJva2Utd2lkdGg6MS4wMzEwMDAwMjtzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjpyb3VuZDtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoMjYiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnMjgiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMTM3Ljg2MzMsNDMuMDIzNCkiPjxwYXRoCiAgICAgICAgICAgICBkPSJtIDAsMCBjIDAsLTE2LjcxMiAtMTMuNTQ4LC0zMC4yNjIgLTMwLjI2MywtMzAuMjYyIC0xNi43MTUsMCAtMzAuMjYzLDEzLjU1IC0zMC4yNjMsMzAuMjYyIDAsMTYuNzE0IDEzLjU0OCwzMC4yNjQgMzAuMjYzLDMwLjI2NCBDIC0xMy41NDgsMzAuMjY0IDAsMTYuNzE0IDAsMCIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOiNmZmZmZmY7ZmlsbC1vcGFjaXR5OjE7ZmlsbC1ydWxlOm5vbnplcm87c3Ryb2tlOm5vbmUiCiAgICAgICAgICAgICBpZD0icGF0aDMwIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzMyIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEzNy44NjMzLDQzLjAyMzQpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAwLC0xNi43MTIgLTEzLjU0OCwtMzAuMjYyIC0zMC4yNjMsLTMwLjI2MiAtMTYuNzE1LDAgLTMwLjI2MywxMy41NSAtMzAuMjYzLDMwLjI2MiAwLDE2LjcxNCAxMy41NDgsMzAuMjY0IDMwLjI2MywzMC4yNjQgQyAtMTMuNTQ4LDMwLjI2NCAwLDE2LjcxNCAwLDAgWiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDM0IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzM2IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEyNi4xNTcyLDEwLjQzMjYpIj48cGF0aAogICAgICAgICAgICAgZD0iTSAwLDAgMTMuNDUyLDYzLjU4NiBIIDI0LjQ1OCBMIDIyLjAxMiwtMS4yMjIgWiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOiNmZmZmZmY7ZmlsbC1vcGFjaXR5OjE7ZmlsbC1ydWxlOm5vbnplcm87c3Ryb2tlOm5vbmUiCiAgICAgICAgICAgICBpZD0icGF0aDM4IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzQwIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDE0MCw3My42NzUzKSI+PHBhdGgKICAgICAgICAgICAgIGQ9Ik0gMCwwIC0xMy4wODYsLTYxLjU1MyIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDQyIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzQ0IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDIwOC43ODcxLDQzLjAyMzQpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAwLC0xNi43MTIgLTEzLjU0OCwtMzAuMjYyIC0zMC4yNjUsLTMwLjI2MiAtMTYuNzEsMCAtMzAuMjYsMTMuNTUgLTMwLjI2LDMwLjI2MiAwLDE2LjcxNCAxMy41NSwzMC4yNjQgMzAuMjYsMzAuMjY0IEMgLTEzLjU0OCwzMC4yNjQgMCwxNi43MTQgMCwwIFoiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDpub25lO3N0cm9rZTojYWExMThlO3N0cm9rZS13aWR0aDozLjgwNTAwMDA3O3N0cm9rZS1saW5lY2FwOmJ1dHQ7c3Ryb2tlLWxpbmVqb2luOm1pdGVyO3N0cm9rZS1taXRlcmxpbWl0OjEwO3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgICAgICAgICAgaWQ9InBhdGg0NiIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc0OCIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNTAuNjE1Miw2NC4yMzczKSI+PHBhdGgKICAgICAgICAgICAgIGQ9Im0gMCwwIDguNTU4LC03LjMzOSAzOS4xMzIsMi40NDcgMTEuMDA2LDMuNjY5IDQuODksLTMxLjc5MyAtMjMuMjMyLC0yNC40NTggSCAxNC42NzMgbCAtMjMuODQ4LDI1LjY3OSB6IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MTtmaWxsLXJ1bGU6bm9uemVybztzdHJva2U6bm9uZSIKICAgICAgICAgICAgIGlkPSJwYXRoNTAiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnNTIiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMTU3LjExOTEsNzMuNjc1MykiPjxwYXRoCiAgICAgICAgICAgICBkPSJNIDAsMCAtMTMuMDgzLC02MS41NTMiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDpub25lO3N0cm9rZTojYWExMThlO3N0cm9rZS13aWR0aDozLjgwNTAwMDA3O3N0cm9rZS1saW5lY2FwOmJ1dHQ7c3Ryb2tlLWxpbmVqb2luOm1pdGVyO3N0cm9rZS1taXRlcmxpbWl0OjEwO3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgICAgICAgICAgaWQ9InBhdGg1NCIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc1NiIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSgzMDIuNDA1MywxMTIuODcwMSkiPjxwYXRoCiAgICAgICAgICAgICBkPSJNIDAsMCAtMjEuNDEzLC0xMDAuNzQ4IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2FhMTE4ZTtzdHJva2Utd2lkdGg6My44MDUwMDAwNztzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjptaXRlcjtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoNTgiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnNjAiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMzIxLjk2ODgsMTEyLjg3MDEpIj48cGF0aAogICAgICAgICAgICAgZD0iTSAwLDAgLTIxLjQxMSwtMTAwLjc0OCIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDYyIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzY0IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDM4Ni4wOTc3LDQzLjAyMzQpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAwLC0xNi43MTIgLTEzLjU0OCwtMzAuMjYyIC0zMC4yNjIsLTMwLjI2MiAtMTYuNzE3LDAgLTMwLjI2NCwxMy41NSAtMzAuMjY0LDMwLjI2MiAwLDE2LjcxNCAxMy41NDcsMzAuMjY0IDMwLjI2NCwzMC4yNjQgQyAtMTMuNTQ4LDMwLjI2NCAwLDE2LjcxNCAwLDAgWiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDY2IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzY4IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDM3My4yODMyLDY4LjM1MjUpIj48cGF0aAogICAgICAgICAgICAgZD0iTSAwLDAgLTM2LjAyOSwtNDkuNTg2IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2FhMTE4ZTtzdHJva2Utd2lkdGg6My44MDUwMDAwNztzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjptaXRlcjtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoNzAiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnNzIiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMjcyLjM3NCw0My4wMjM0KSI+PHBhdGgKICAgICAgICAgICAgIGQ9Im0gMCwwIGMgMCwtMTYuNzEyIC0xMy41NDksLTMwLjI2MiAtMzAuMjYzLC0zMC4yNjIgLTE2LjcxMywwIC0zMC4yNjEsMTMuNTUgLTMwLjI2MSwzMC4yNjIgMCwxNi43MTQgMTMuNTQ4LDMwLjI2NCAzMC4yNjEsMzAuMjY0IEMgLTEzLjU0OSwzMC4yNjQgMCwxNi43MTQgMCwwIFoiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDpub25lO3N0cm9rZTojYWExMThlO3N0cm9rZS13aWR0aDozLjgwNTAwMDA3O3N0cm9rZS1saW5lY2FwOmJ1dHQ7c3Ryb2tlLWxpbmVqb2luOm1pdGVyO3N0cm9rZS1taXRlcmxpbWl0OjEwO3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgICAgICAgICAgaWQ9InBhdGg3NCIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc3NiIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSgyNTkuNTU5Niw2OC4zNTI1KSI+PHBhdGgKICAgICAgICAgICAgIGQ9Ik0gMCwwIC0zNi4wMjYsLTQ5LjU4NiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDc4IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzgwIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDI3NC43MzM0LDE3Ljc3MDUpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgLTE4LjM0Miw2LjExNCAxLjgzNiwzOC43MjEgNC44OSw3LjU0MiAxNC4wNjIsLTEyLjAyNCB6IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MTtmaWxsLXJ1bGU6bm9uemVybztzdHJva2U6bm9uZSIKICAgICAgICAgICAgIGlkPSJwYXRoODIiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnODQiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMzg4LjI1MjksMTcuNTY1OSkiPjxwYXRoCiAgICAgICAgICAgICBkPSJtIDAsMCAtMTguMzQzLDYuMTE0IDEuODM3LDM4LjcyMSA0Ljg4OSw3LjU0MyAxNC4wNjMsLTEyLjAyNCB6IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MTtmaWxsLXJ1bGU6bm9uemVybztzdHJva2U6bm9uZSIKICAgICAgICAgICAgIGlkPSJwYXRoODYiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnODgiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoOTcuMjYwNyw3Mi44NjUyKSI+PHBhdGgKICAgICAgICAgICAgIGQ9Im0gMCwwIGMgLTIuMTIxLDIuNzgyIC04LjUzOSw1LjIzOSAtMTEuMzksMTUuNTIxIC0xLjUxNCw1LjQ1OCAtMC4zNDgsOS4wNDggMC42OTcsMTMuOTIgMS4xMTcsNS4yMDUgLTIuMDg4LDguMDA0IC02Ljk2MSw3LjY1NiAtOS43NDQsLTAuNjk2IC0zNS41NTksLTYuNDE4IC02Ni4xMjIsNS41NjkgLTExLjk1Niw0LjY4OCAtMTAuNzY1LDQuMDY4IC03LjYzLC0zLjc1NSA0Ljk4NCwtMTIuNDM5IDI2LjUwNywtMjEuNzg5IDQ1LjE1MiwtMjIuNzc3IC0xMC4xMDYsMC45MDkgLTIyLjE0LDMuMTI5IC0yNS41MDIsNC4wMDEgLTEzLjUsMy41IDcsLTEzLjUgMzQuNTAyLC0xNC4yNTEgLTcuNDg0LDAuMjM3IC0xOS41OTgsMS40ODYgLTIyLjUsMi4yNSAtMTksNSA0LC0xMS45OTkgMzIuMjUsLTEzLjUwMiA0LjQzNiwtMC4yMzYgMTUuNzMxLC02Ljg5MiAyMS44NjksLTMuMjk1IDMuODM0LDIuMjQ2IDEuMzc5LDAuODAxIDMuODM0LDIuMjQ2IEMgMy44ODgsLTMuMDY4IDAsMCAwLDAiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDojNzdkMWM4O2ZpbGwtb3BhY2l0eToxO2ZpbGwtcnVsZTpub256ZXJvO3N0cm9rZTpub25lIgogICAgICAgICAgICAgaWQ9InBhdGg5MCIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc5MiIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSg5Ny4yNjA3LDcyLjg2NTIpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAtMi4xMjEsMi43ODIgLTguNTM5LDUuMjM5IC0xMS4zOSwxNS41MjEgLTEuNTE0LDUuNDU4IC0wLjM0OCw5LjA0OCAwLjY5NywxMy45MiAxLjExNyw1LjIwNSAtMi4wODgsOC4wMDQgLTYuOTYxLDcuNjU2IC05Ljc0NCwtMC42OTYgLTM1LjU1OSwtNi40MTggLTY2LjEyMiw1LjU2OSAtMTEuOTU2LDQuNjg4IC0xMC43NjUsNC4wNjggLTcuNjMsLTMuNzU1IDQuOTg0LC0xMi40MzkgMjYuNTA3LC0yMS43ODkgNDUuMTUyLC0yMi43NzcgLTEwLjEwNiwwLjkwOSAtMjIuMTQsMy4xMjkgLTI1LjUwMiw0LjAwMSAtMTMuNSwzLjUgNywtMTMuNSAzNC41MDIsLTE0LjI1MSAtNy40ODQsMC4yMzcgLTE5LjU5OCwxLjQ4NiAtMjIuNSwyLjI1IC0xOSw1IDQsLTExLjk5OSAzMi4yNSwtMTMuNTAyIDQuNDM2LC0wLjIzNiAxNS43MzEsLTYuODkyIDIxLjg2OSwtMy4yOTUgMy44MzQsMi4yNDYgMS4zNzksMC44MDEgMy44MzQsMi4yNDYgQyAzLjg4OCwtMy4wNjggMCwwIDAsMCBaIgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2ZmZmZmZjtzdHJva2Utd2lkdGg6MjtzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjpyb3VuZDtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoOTQiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48L2c+PC9nPjx0ZXh0CiAgICAgICB0cmFuc2Zvcm09Im1hdHJpeCgxLDAsMCwtMSwzODkuNDk5LDYyLjk5NjEpIgogICAgICAgc3R5bGU9ImZvbnQtdmFyaWFudDpub3JtYWw7Zm9udC13ZWlnaHQ6bm9ybWFsO2ZvbnQtc3RyZXRjaDpub3JtYWw7Zm9udC1zaXplOjE4cHg7Zm9udC1mYW1pbHk6J1RyZWJ1Y2hldCBNUyc7LWlua3NjYXBlLWZvbnQtc3BlY2lmaWNhdGlvbjpUcmVidWNoZXRNUzt3cml0aW5nLW1vZGU6bHItdGI7ZmlsbDojYWExMThlO2ZpbGwtb3BhY2l0eToxO2ZpbGwtcnVsZTpub256ZXJvO3N0cm9rZTpub25lIgogICAgICAgaWQ9InRleHQ5OCI+PHRzcGFuCiAgICAgICAgIHg9IjAiCiAgICAgICAgIHk9IjAiCiAgICAgICAgIGlkPSJ0c3Bhbjk2Ij7CrjwvdHNwYW4+PC90ZXh0Pgo8L2c+PC9zdmc+#svgView(preserveAspectRatio(none))'
```

it will fail with a:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sagesmith/.pyenv/versions/3.11.0/lib/python3.11/base64.py", line 88, in b64decode
    return binascii.a2b_base64(s, strict_mode=validate)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
binascii.Error: Incorrect padding
```

but if I add a newline to the end for example and do the same thing, there is no error

```python
image = 'PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjwhLS0gQ3JlYXRlZCB3aXRoIElua3NjYXBlIChodHRwOi8vd3d3Lmlua3NjYXBlLm9yZy8pIC0tPgoKPHN2ZwogICB4bWxuczpkYz0iaHR0cDovL3B1cmwub3JnL2RjL2VsZW1lbnRzLzEuMS8iCiAgIHhtbG5zOmNjPSJodHRwOi8vY3JlYXRpdmVjb21tb25zLm9yZy9ucyMiCiAgIHhtbG5zOnJkZj0iaHR0cDovL3d3dy53My5vcmcvMTk5OS8wMi8yMi1yZGYtc3ludGF4LW5zIyIKICAgeG1sbnM6c3ZnPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIKICAgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIgogICB4bWxuczpzb2RpcG9kaT0iaHR0cDovL3NvZGlwb2RpLnNvdXJjZWZvcmdlLm5ldC9EVEQvc29kaXBvZGktMC5kdGQiCiAgIHhtbG5zOmlua3NjYXBlPSJodHRwOi8vd3d3Lmlua3NjYXBlLm9yZy9uYW1lc3BhY2VzL2lua3NjYXBlIgogICB2ZXJzaW9uPSIxLjEiCiAgIGlkPSJzdmcyIgogICB4bWw6c3BhY2U9InByZXNlcnZlIgogICB3aWR0aD0iNTM2LjM2NDc1IgogICBoZWlnaHQ9IjE1MC4zMTAwMyIKICAgdmlld0JveD0iMCAwIDUzNi4zNjQ3MyAxNTAuMzEwMDMiCiAgIHNvZGlwb2RpOmRvY25hbWU9ImFyZWxsZS1sb2dvLnN2ZyIKICAgaW5rc2NhcGU6dmVyc2lvbj0iMC45Mi41ICgyMDYwZWMxZjlmLCAyMDIwLTA0LTA4KSI+PG1ldGFkYXRhCiAgICAgaWQ9Im1ldGFkYXRhOCI+PHJkZjpSREY+PGNjOldvcmsKICAgICAgICAgcmRmOmFib3V0PSIiPjxkYzpmb3JtYXQ+aW1hZ2Uvc3ZnK3htbDwvZGM6Zm9ybWF0PjxkYzp0eXBlCiAgICAgICAgICAgcmRmOnJlc291cmNlPSJodHRwOi8vcHVybC5vcmcvZGMvZGNtaXR5cGUvU3RpbGxJbWFnZSIgLz48ZGM6dGl0bGU+PC9kYzp0aXRsZT48L2NjOldvcms+PC9yZGY6UkRGPjwvbWV0YWRhdGE+PGRlZnMKICAgICBpZD0iZGVmczYiPjxjbGlwUGF0aAogICAgICAgY2xpcFBhdGhVbml0cz0idXNlclNwYWNlT25Vc2UiCiAgICAgICBpZD0iY2xpcFBhdGgxOCI+PHBhdGgKICAgICAgICAgZD0iTSAwLDEyMiBIIDc2OC41MDUgViAwIEggMCBaIgogICAgICAgICBpZD0icGF0aDE2IgogICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvY2xpcFBhdGg+PC9kZWZzPjxzb2RpcG9kaTpuYW1lZHZpZXcKICAgICBwYWdlY29sb3I9IiNmZmZmZmYiCiAgICAgYm9yZGVyY29sb3I9IiM2NjY2NjYiCiAgICAgYm9yZGVyb3BhY2l0eT0iMSIKICAgICBvYmplY3R0b2xlcmFuY2U9IjEwIgogICAgIGdyaWR0b2xlcmFuY2U9IjEwIgogICAgIGd1aWRldG9sZXJhbmNlPSIxMCIKICAgICBpbmtzY2FwZTpwYWdlb3BhY2l0eT0iMCIKICAgICBpbmtzY2FwZTpwYWdlc2hhZG93PSIyIgogICAgIGlua3NjYXBlOndpbmRvdy13aWR0aD0iMTg2NCIKICAgICBpbmtzY2FwZTp3aW5kb3ctaGVpZ2h0PSIxMDI0IgogICAgIGlkPSJuYW1lZHZpZXc0IgogICAgIHNob3dncmlkPSJmYWxzZSIKICAgICBpbmtzY2FwZTp6b29tPSIwLjcyMDIyOTUzIgogICAgIGlua3NjYXBlOmN4PSI3ODguNzIwNTgiCiAgICAgaW5rc2NhcGU6Y3k9Ii04NC40NDIyMDQiCiAgICAgaW5rc2NhcGU6d2luZG93LXg9IjU2IgogICAgIGlua3NjYXBlOndpbmRvdy15PSIyNyIKICAgICBpbmtzY2FwZTp3aW5kb3ctbWF4aW1pemVkPSIxIgogICAgIGlua3NjYXBlOmN1cnJlbnQtbGF5ZXI9ImcxMCIKICAgICBmaXQtbWFyZ2luLXRvcD0iMCIKICAgICBmaXQtbWFyZ2luLWxlZnQ9IjAiCiAgICAgZml0LW1hcmdpbi1yaWdodD0iMCIKICAgICBmaXQtbWFyZ2luLWJvdHRvbT0iMCIgLz48ZwogICAgIGlkPSJnMTAiCiAgICAgaW5rc2NhcGU6Z3JvdXBtb2RlPSJsYXllciIKICAgICBpbmtzY2FwZTpsYWJlbD0iYXJlbGxlLWxvZ28iCiAgICAgdHJhbnNmb3JtPSJtYXRyaXgoMS4zMzMzMzMzLDAsMCwtMS4zMzMzMzMzLC0zLjY1MDgxNDEsMTU5LjMyNzc3KSI+PGcKICAgICAgIGlkPSJnMTIiPjxnCiAgICAgICAgIGlkPSJnMTQiCiAgICAgICAgIGNsaXAtcGF0aD0idXJsKCNjbGlwUGF0aDE4KSI+PGcKICAgICAgICAgICBpZD0iZzIwIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEwOS42Njg5LDY3LjE2MzEpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAtMS41OTgsMi42NTMgLTcuMjkyLDUuOTg3IC04LjgwOCwxNS4zMDcgLTAuODA0LDQuOTQ5IDAuNTY5LDcuOTk0IDEuOTU1LDEyLjE4MSAxLjQ4OCw0LjQ3MSAtMS4wNTksNy4yNDQgLTUuMzgsNy40MDggLTguNjQ0LDAuMzMgLTMxLjkwNywtMi4yMTEgLTU3LjYzOCwxMS4yODggLTEwLjA2Niw1LjI4IC0xMC43NTksMy45MjIgLTcuMDc2LC0yLjU2NiA1LjY0NSwtOS45NTIgMTkuMzc4LC0xNS4zMzQgMzUuNjg3LC0xOC4wMDUgLTQ1LjM5Nyw5LjgyNSAtMTcuMTY2LC0yLjE3NCA1LjA5OCwtNy4xMjcgLTQxLjA1Niw4LjQ4NCAtMTUuNjI1LC0zLjAzNSA0LjQ0OCwtNi43MTEgLTM4LjQ3NSw2LjgyMSAtMTEuMzg3LC0yLjk3OCAxLjQ3NywtNS40MjggMy44NjIsLTAuNzM2IDEyLjM5MSwtMTMuMDI2IDE4LjA1NCwtMTAuMjczIDYuMzE3LDMuMDY5IDQuNDg4LDAuNjg5IDcuNDc5LDIuMDI2IEMgMC41NjYsMC40NTggMCwwIDAsMCIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOiNhYTExOGU7ZmlsbC1vcGFjaXR5OjE7ZmlsbC1ydWxlOm5vbnplcm87c3Ryb2tlOm5vbmUiCiAgICAgICAgICAgICBpZD0icGF0aDIyIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzI0IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEwOS42Njg5LDY3LjE2MzEpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAtMS41OTgsMi42NTMgLTcuMjkyLDUuOTg3IC04LjgwOCwxNS4zMDcgLTAuODA0LDQuOTQ5IDAuNTY5LDcuOTk0IDEuOTU1LDEyLjE4MSAxLjQ4OCw0LjQ3MSAtMS4wNTksNy4yNDQgLTUuMzgsNy40MDggLTguNjQ0LDAuMzMgLTMxLjkwNywtMi4yMTEgLTU3LjYzOCwxMS4yODggLTEwLjA2Niw1LjI4IC0xMC43NTksMy45MjIgLTcuMDc2LC0yLjU2NiA1LjY0NSwtOS45NTIgMTkuMzc4LC0xNS4zMzQgMzUuNjg3LC0xOC4wMDUgLTQ1LjM5Nyw5LjgyNSAtMTcuMTY2LC0yLjE3NCA1LjA5OCwtNy4xMjcgLTQxLjA1Niw4LjQ4NCAtMTUuNjI1LC0zLjAzNSA0LjQ0OCwtNi43MTEgLTM4LjQ3NSw2LjgyMSAtMTEuMzg3LC0yLjk3OCAxLjQ3NywtNS40MjggMy44NjIsLTAuNzM2IDEyLjM5MSwtMTMuMDI2IDE4LjA1NCwtMTAuMjczIDYuMzE3LDMuMDY5IDQuNDg4LDAuNjg5IDcuNDc5LDIuMDI2IEMgMC41NjYsMC40NTggMCwwIDAsMCBaIgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2ZmZmZmZjtzdHJva2Utd2lkdGg6MS4wMzEwMDAwMjtzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjpyb3VuZDtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoMjYiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnMjgiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMTM3Ljg2MzMsNDMuMDIzNCkiPjxwYXRoCiAgICAgICAgICAgICBkPSJtIDAsMCBjIDAsLTE2LjcxMiAtMTMuNTQ4LC0zMC4yNjIgLTMwLjI2MywtMzAuMjYyIC0xNi43MTUsMCAtMzAuMjYzLDEzLjU1IC0zMC4yNjMsMzAuMjYyIDAsMTYuNzE0IDEzLjU0OCwzMC4yNjQgMzAuMjYzLDMwLjI2NCBDIC0xMy41NDgsMzAuMjY0IDAsMTYuNzE0IDAsMCIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOiNmZmZmZmY7ZmlsbC1vcGFjaXR5OjE7ZmlsbC1ydWxlOm5vbnplcm87c3Ryb2tlOm5vbmUiCiAgICAgICAgICAgICBpZD0icGF0aDMwIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzMyIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEzNy44NjMzLDQzLjAyMzQpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAwLC0xNi43MTIgLTEzLjU0OCwtMzAuMjYyIC0zMC4yNjMsLTMwLjI2MiAtMTYuNzE1LDAgLTMwLjI2MywxMy41NSAtMzAuMjYzLDMwLjI2MiAwLDE2LjcxNCAxMy41NDgsMzAuMjY0IDMwLjI2MywzMC4yNjQgQyAtMTMuNTQ4LDMwLjI2NCAwLDE2LjcxNCAwLDAgWiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDM0IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzM2IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDEyNi4xNTcyLDEwLjQzMjYpIj48cGF0aAogICAgICAgICAgICAgZD0iTSAwLDAgMTMuNDUyLDYzLjU4NiBIIDI0LjQ1OCBMIDIyLjAxMiwtMS4yMjIgWiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOiNmZmZmZmY7ZmlsbC1vcGFjaXR5OjE7ZmlsbC1ydWxlOm5vbnplcm87c3Ryb2tlOm5vbmUiCiAgICAgICAgICAgICBpZD0icGF0aDM4IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzQwIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDE0MCw3My42NzUzKSI+PHBhdGgKICAgICAgICAgICAgIGQ9Ik0gMCwwIC0xMy4wODYsLTYxLjU1MyIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDQyIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzQ0IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDIwOC43ODcxLDQzLjAyMzQpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAwLC0xNi43MTIgLTEzLjU0OCwtMzAuMjYyIC0zMC4yNjUsLTMwLjI2MiAtMTYuNzEsMCAtMzAuMjYsMTMuNTUgLTMwLjI2LDMwLjI2MiAwLDE2LjcxNCAxMy41NSwzMC4yNjQgMzAuMjYsMzAuMjY0IEMgLTEzLjU0OCwzMC4yNjQgMCwxNi43MTQgMCwwIFoiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDpub25lO3N0cm9rZTojYWExMThlO3N0cm9rZS13aWR0aDozLjgwNTAwMDA3O3N0cm9rZS1saW5lY2FwOmJ1dHQ7c3Ryb2tlLWxpbmVqb2luOm1pdGVyO3N0cm9rZS1taXRlcmxpbWl0OjEwO3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgICAgICAgICAgaWQ9InBhdGg0NiIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc0OCIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSgxNTAuNjE1Miw2NC4yMzczKSI+PHBhdGgKICAgICAgICAgICAgIGQ9Im0gMCwwIDguNTU4LC03LjMzOSAzOS4xMzIsMi40NDcgMTEuMDA2LDMuNjY5IDQuODksLTMxLjc5MyAtMjMuMjMyLC0yNC40NTggSCAxNC42NzMgbCAtMjMuODQ4LDI1LjY3OSB6IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MTtmaWxsLXJ1bGU6bm9uemVybztzdHJva2U6bm9uZSIKICAgICAgICAgICAgIGlkPSJwYXRoNTAiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnNTIiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMTU3LjExOTEsNzMuNjc1MykiPjxwYXRoCiAgICAgICAgICAgICBkPSJNIDAsMCAtMTMuMDgzLC02MS41NTMiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDpub25lO3N0cm9rZTojYWExMThlO3N0cm9rZS13aWR0aDozLjgwNTAwMDA3O3N0cm9rZS1saW5lY2FwOmJ1dHQ7c3Ryb2tlLWxpbmVqb2luOm1pdGVyO3N0cm9rZS1taXRlcmxpbWl0OjEwO3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgICAgICAgICAgaWQ9InBhdGg1NCIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc1NiIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSgzMDIuNDA1MywxMTIuODcwMSkiPjxwYXRoCiAgICAgICAgICAgICBkPSJNIDAsMCAtMjEuNDEzLC0xMDAuNzQ4IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2FhMTE4ZTtzdHJva2Utd2lkdGg6My44MDUwMDAwNztzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjptaXRlcjtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoNTgiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnNjAiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMzIxLjk2ODgsMTEyLjg3MDEpIj48cGF0aAogICAgICAgICAgICAgZD0iTSAwLDAgLTIxLjQxMSwtMTAwLjc0OCIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDYyIgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzY0IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDM4Ni4wOTc3LDQzLjAyMzQpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAwLC0xNi43MTIgLTEzLjU0OCwtMzAuMjYyIC0zMC4yNjIsLTMwLjI2MiAtMTYuNzE3LDAgLTMwLjI2NCwxMy41NSAtMzAuMjY0LDMwLjI2MiAwLDE2LjcxNCAxMy41NDcsMzAuMjY0IDMwLjI2NCwzMC4yNjQgQyAtMTMuNTQ4LDMwLjI2NCAwLDE2LjcxNCAwLDAgWiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDY2IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzY4IgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDM3My4yODMyLDY4LjM1MjUpIj48cGF0aAogICAgICAgICAgICAgZD0iTSAwLDAgLTM2LjAyOSwtNDkuNTg2IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2FhMTE4ZTtzdHJva2Utd2lkdGg6My44MDUwMDAwNztzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjptaXRlcjtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoNzAiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnNzIiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMjcyLjM3NCw0My4wMjM0KSI+PHBhdGgKICAgICAgICAgICAgIGQ9Im0gMCwwIGMgMCwtMTYuNzEyIC0xMy41NDksLTMwLjI2MiAtMzAuMjYzLC0zMC4yNjIgLTE2LjcxMywwIC0zMC4yNjEsMTMuNTUgLTMwLjI2MSwzMC4yNjIgMCwxNi43MTQgMTMuNTQ4LDMwLjI2NCAzMC4yNjEsMzAuMjY0IEMgLTEzLjU0OSwzMC4yNjQgMCwxNi43MTQgMCwwIFoiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDpub25lO3N0cm9rZTojYWExMThlO3N0cm9rZS13aWR0aDozLjgwNTAwMDA3O3N0cm9rZS1saW5lY2FwOmJ1dHQ7c3Ryb2tlLWxpbmVqb2luOm1pdGVyO3N0cm9rZS1taXRlcmxpbWl0OjEwO3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgICAgICAgICAgaWQ9InBhdGg3NCIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc3NiIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSgyNTkuNTU5Niw2OC4zNTI1KSI+PHBhdGgKICAgICAgICAgICAgIGQ9Ik0gMCwwIC0zNi4wMjYsLTQ5LjU4NiIKICAgICAgICAgICAgIHN0eWxlPSJmaWxsOm5vbmU7c3Ryb2tlOiNhYTExOGU7c3Ryb2tlLXdpZHRoOjMuODA1MDAwMDc7c3Ryb2tlLWxpbmVjYXA6YnV0dDtzdHJva2UtbGluZWpvaW46bWl0ZXI7c3Ryb2tlLW1pdGVybGltaXQ6MTA7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgICAgICAgICBpZD0icGF0aDc4IgogICAgICAgICAgICAgaW5rc2NhcGU6Y29ubmVjdG9yLWN1cnZhdHVyZT0iMCIgLz48L2c+PGcKICAgICAgICAgICBpZD0iZzgwIgogICAgICAgICAgIHRyYW5zZm9ybT0idHJhbnNsYXRlKDI3NC43MzM0LDE3Ljc3MDUpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgLTE4LjM0Miw2LjExNCAxLjgzNiwzOC43MjEgNC44OSw3LjU0MiAxNC4wNjIsLTEyLjAyNCB6IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MTtmaWxsLXJ1bGU6bm9uemVybztzdHJva2U6bm9uZSIKICAgICAgICAgICAgIGlkPSJwYXRoODIiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnODQiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMzg4LjI1MjksMTcuNTY1OSkiPjxwYXRoCiAgICAgICAgICAgICBkPSJtIDAsMCAtMTguMzQzLDYuMTE0IDEuODM3LDM4LjcyMSA0Ljg4OSw3LjU0MyAxNC4wNjMsLTEyLjAyNCB6IgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6I2ZmZmZmZjtmaWxsLW9wYWNpdHk6MTtmaWxsLXJ1bGU6bm9uemVybztzdHJva2U6bm9uZSIKICAgICAgICAgICAgIGlkPSJwYXRoODYiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48ZwogICAgICAgICAgIGlkPSJnODgiCiAgICAgICAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoOTcuMjYwNyw3Mi44NjUyKSI+PHBhdGgKICAgICAgICAgICAgIGQ9Im0gMCwwIGMgLTIuMTIxLDIuNzgyIC04LjUzOSw1LjIzOSAtMTEuMzksMTUuNTIxIC0xLjUxNCw1LjQ1OCAtMC4zNDgsOS4wNDggMC42OTcsMTMuOTIgMS4xMTcsNS4yMDUgLTIuMDg4LDguMDA0IC02Ljk2MSw3LjY1NiAtOS43NDQsLTAuNjk2IC0zNS41NTksLTYuNDE4IC02Ni4xMjIsNS41NjkgLTExLjk1Niw0LjY4OCAtMTAuNzY1LDQuMDY4IC03LjYzLC0zLjc1NSA0Ljk4NCwtMTIuNDM5IDI2LjUwNywtMjEuNzg5IDQ1LjE1MiwtMjIuNzc3IC0xMC4xMDYsMC45MDkgLTIyLjE0LDMuMTI5IC0yNS41MDIsNC4wMDEgLTEzLjUsMy41IDcsLTEzLjUgMzQuNTAyLC0xNC4yNTEgLTcuNDg0LDAuMjM3IC0xOS41OTgsMS40ODYgLTIyLjUsMi4yNSAtMTksNSA0LC0xMS45OTkgMzIuMjUsLTEzLjUwMiA0LjQzNiwtMC4yMzYgMTUuNzMxLC02Ljg5MiAyMS44NjksLTMuMjk1IDMuODM0LDIuMjQ2IDEuMzc5LDAuODAxIDMuODM0LDIuMjQ2IEMgMy44ODgsLTMuMDY4IDAsMCAwLDAiCiAgICAgICAgICAgICBzdHlsZT0iZmlsbDojNzdkMWM4O2ZpbGwtb3BhY2l0eToxO2ZpbGwtcnVsZTpub256ZXJvO3N0cm9rZTpub25lIgogICAgICAgICAgICAgaWQ9InBhdGg5MCIKICAgICAgICAgICAgIGlua3NjYXBlOmNvbm5lY3Rvci1jdXJ2YXR1cmU9IjAiIC8+PC9nPjxnCiAgICAgICAgICAgaWQ9Imc5MiIKICAgICAgICAgICB0cmFuc2Zvcm09InRyYW5zbGF0ZSg5Ny4yNjA3LDcyLjg2NTIpIj48cGF0aAogICAgICAgICAgICAgZD0ibSAwLDAgYyAtMi4xMjEsMi43ODIgLTguNTM5LDUuMjM5IC0xMS4zOSwxNS41MjEgLTEuNTE0LDUuNDU4IC0wLjM0OCw5LjA0OCAwLjY5NywxMy45MiAxLjExNyw1LjIwNSAtMi4wODgsOC4wMDQgLTYuOTYxLDcuNjU2IC05Ljc0NCwtMC42OTYgLTM1LjU1OSwtNi40MTggLTY2LjEyMiw1LjU2OSAtMTEuOTU2LDQuNjg4IC0xMC43NjUsNC4wNjggLTcuNjMsLTMuNzU1IDQuOTg0LC0xMi40MzkgMjYuNTA3LC0yMS43ODkgNDUuMTUyLC0yMi43NzcgLTEwLjEwNiwwLjkwOSAtMjIuMTQsMy4xMjkgLTI1LjUwMiw0LjAwMSAtMTMuNSwzLjUgNywtMTMuNSAzNC41MDIsLTE0LjI1MSAtNy40ODQsMC4yMzcgLTE5LjU5OCwxLjQ4NiAtMjIuNSwyLjI1IC0xOSw1IDQsLTExLjk5OSAzMi4yNSwtMTMuNTAyIDQuNDM2LC0wLjIzNiAxNS43MzEsLTYuODkyIDIxLjg2OSwtMy4yOTUgMy44MzQsMi4yNDYgMS4zNzksMC44MDEgMy44MzQsMi4yNDYgQyAzLjg4OCwtMy4wNjggMCwwIDAsMCBaIgogICAgICAgICAgICAgc3R5bGU9ImZpbGw6bm9uZTtzdHJva2U6I2ZmZmZmZjtzdHJva2Utd2lkdGg6MjtzdHJva2UtbGluZWNhcDpidXR0O3N0cm9rZS1saW5lam9pbjpyb3VuZDtzdHJva2UtbWl0ZXJsaW1pdDoxMDtzdHJva2UtZGFzaGFycmF5Om5vbmU7c3Ryb2tlLW9wYWNpdHk6MSIKICAgICAgICAgICAgIGlkPSJwYXRoOTQiCiAgICAgICAgICAgICBpbmtzY2FwZTpjb25uZWN0b3ItY3VydmF0dXJlPSIwIiAvPjwvZz48L2c+PC9nPjx0ZXh0CiAgICAgICB0cmFuc2Zvcm09Im1hdHJpeCgxLDAsMCwtMSwzODkuNDk5LDYyLjk5NjEpIgogICAgICAgc3R5bGU9ImZvbnQtdmFyaWFudDpub3JtYWw7Zm9udC13ZWlnaHQ6bm9ybWFsO2ZvbnQtc3RyZXRjaDpub3JtYWw7Zm9udC1zaXplOjE4cHg7Zm9udC1mYW1pbHk6J1RyZWJ1Y2hldCBNUyc7LWlua3NjYXBlLWZvbnQtc3BlY2lmaWNhdGlvbjpUcmVidWNoZXRNUzt3cml0aW5nLW1vZGU6bHItdGI7ZmlsbDojYWExMThlO2ZpbGwtb3BhY2l0eToxO2ZpbGwtcnVsZTpub256ZXJvO3N0cm9rZTpub25lIgogICAgICAgaWQ9InRleHQ5OCI+PHRzcGFuCiAgICAgICAgIHg9IjAiCiAgICAgICAgIHk9IjAiCiAgICAgICAgIGlkPSJ0c3Bhbjk2Ij7CrjwvdHNwYW4+PC90ZXh0Pgo8L2c+PC9zdmc+Cg==#svgView(preserveAspectRatio(none))'
```

the base64 library appears to split on `=` (padding) so images that include padding are allowed to include fragments while images that don't include padding are unconditionally not allowed. Perhaps something like might fix the problem:

```python
r'data:image([^,;]*)(;base64)?,(.*)(#.*)?$'
```

Reply:
> I think that ESEF is more restrictive than bare SVG.

Can you elaborate?  The fragments aren't "executable code", they're just style-like transforms of the rendering of the image, comparable to CSS image transforms.  It doesn't seem like ESEF has (or should have) an opinion on that.

Reply:
I have checked the ESEF reporting manual and as a matter of fact, there is no restriction regarding URI fragments. Formally, SVG fragments are just URI fragments. Consequently they are allowed! 

Thus, it would be useful to change the code as suggested by you.

By the way, does it make sense to also allow query parameters?

Reply:
> By the way, does it make sense to also allow query parameters?

I've never seen that, and my brief look at the RFCs didn't find anything.  I did find that [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs#no_support_for_query_strings_etc) states they are not supported (i.e., the `?` is considered raw data like any other character), and [this comment](https://bugs.chromium.org/p/chromium/issues/detail?id=123004#c21) on the Chrome bug for fixing fragments on data URIs says the `?` is part of the raw data (the username is truncated, but it's probably Anne van Kesteren).

Reply:
Thank you for the cross checking!

I have seen that #657 has been opened to cover the SVG fragment case.

Nothing needs to be changed here.

Reply:
The changes from #657 have been integrated.